src/BCReader.jl

```diff
@@ -0,0 +1,309 @@
+    all_dates::D                    # vector of all dates contained in the original data file
+    segment_idx::Vector{Int}        # index of the monthly data in the file
+    segment_idx0::Vector{Int}       # `segment_idx` of the file data that is closest to date0
+    monthly_fields::C               # tuple of the two monthly fields, that will be used for the daily interpolation
+    segment_length::Vector{Int}     # length of each month segment (used in the daily interpolation)
+    scaling_function::O             # function that scales, offsets or transforms the raw variable
+    interpolate_daily::Bool         # switch to trigger daily interpolation
+    land_mask::M                    # mask with 1 = land, 0 = ocean / sea-ice
+"""
+struct BCFileInfo{F, B, X, S, V, D, C, O, M}
+    FT::F
+    bcfile_dir::B
+    comms_ctx::X
+    hd_outfile_root::S
+    varname::V
+    all_dates::D
+    segment_idx::Vector{Int}
+    segment_idx0::Vector{Int}
+    monthly_fields::C
+    segment_length::Vector{Int}
+    scaling_function::O
+    interpolate_daily::Bool
+    land_mask::M
+end
+
+"""
+    no_scaling(field, bcf_info)
+
+Remap the values of a field onto the space of the
+BCFileInfo's land_mask without scaling.
+
+# Arguments
+- `field`: [Fields.Field] contains the values to be remapped.
+- `bcf_info`: [BCFileInfo] contains a land_mask to remap onto the space of.
+"""
+no_scaling(field::Fields.Field, bcf_info::BCFileInfo) = Utilities.swap_space!(field, axes(bcf_info.land_mask))
+
+"""
+    bcfile_info_init(
+        FT,
+        datafile_rll,
+        varname,
+        boundary_space;
+        comms_ctx = ClimaComms.SingletonCommsContext(),
+        interpolate_daily = false,
+        segment_idx0 = nothing,
+        scaling_function = no_scaling,
+        land_mask = nothing,
+        date0 = nothing,
+        mono = true,
+    )
+
+Regrids from lat-lon grid to cgll grid, saving the output in a new file,
+and returns the info packaged in a single struct.
+
+# Arguments
+- `FT`: [DataType] Float type.
+- `bcfile_dir`: [String] directory the BC file is stored in.
+- `datafile_rll`: [String] file containing data to regrid.
+- `varname`: [String] name of the variable to be regridded.
+- `boundary_space`: [Spaces.AbstractSpace] the space to which we are mapping.
+- `comms_ctx`: [ClimaComms.AbstractCommsContext] context used for this operation.
+- `interpolate_daily`: [Bool] switch to trigger daily interpolation.
+- `segment_idx0`: [Vector{Int}] index of the file data that is closest to date0.
+- `scaling function`: [Function] scales, offsets or transforms `varname`.
+- `land_mask`: [Fields.field] mask with 1 = land, 0 = ocean / sea-ice.
+- `date0`: [Dates.DateTime] start date of the file data.
+- `mono`: [Bool] flag for monotone remapping of `datafile_rll`.
+
+# Returns
+- BCFileInfo
+"""
+function bcfile_info_init(
+    FT,
+    bcfile_dir,
+    datafile_rll,
+    varname,
+    boundary_space;
+    comms_ctx = ClimaComms.SingletonCommsContext(),
```

Comment:
```suggestion
    comms_ctx,
```
I think that creating separate `comms_ctx` may lead to strange mpi errors, so I'd suggesting making this a required argument